internal/generate/clusterserviceversion/clusterserviceversion_updaters.go

```diff
@@ -316,16 +336,80 @@ func mutatingToWebhookDescription(webhook admissionregv1.MutatingWebhook) operat
 		AdmissionReviewVersions: webhook.AdmissionReviewVersions,
 		ReinvocationPolicy:      webhook.ReinvocationPolicy,
 	}
+	if len(description.AdmissionReviewVersions) == 0 {
+		description.AdmissionReviewVersions = defaultAdmissionReviewVersions
+	}
+
 	if serviceRef := webhook.ClientConfig.Service; serviceRef != nil {
 		if serviceRef.Port != nil {
 			description.ContainerPort = *serviceRef.Port
 		}
-		description.DeploymentName = strings.TrimSuffix(serviceRef.Name, "-service")
+		description.DeploymentName = depName
+		if description.DeploymentName == "" {
+			description.DeploymentName = strings.TrimSuffix(serviceRef.Name, "-service")
+		}
 		description.WebhookPath = serviceRef.Path
 	}
 	return description
 }
 
+// findMatchingDeploymentAndServiceForWebhook matches a Service to a webhook's client config (if it uses a service)
+// then matches that Service to a Deployment by comparing label selectors (if the Service uses label selectors).
+// The names of both Service and Deployment are returned if found.
+func findMatchingDeploymentAndServiceForWebhook(c *collector.Manifests, wcc admissionregv1.WebhookClientConfig) (depName, serviceName string) {
```

Comment:
I am not sure if this is in scope of this PR. But currently, if using WebHooks we need not specify `volumeMounts` and webhook server port in CSV. Could we have a check for that, and accordingly not append [ContainerVolumeMounts](withContainerVolumeMounts) in Deployment.

Reply:
I think this will need discussion before changing. We can’t remove all volume mounts and/or ports since these will probably contain other values set by users; in general the generator should avoid modifying declarative manifests like that, which risks changing what they declare.

Instead we need to properly patch the deployment via kustomize before passing it to the generator. @joelanford had a thought or two on this.